source/common/stats/thread_local_store.cc

```diff
@@ -137,6 +138,18 @@ void ThreadLocalStoreImpl::initializeThreading(Event::Dispatcher& main_thread_di
   tls_->set([](Event::Dispatcher&) -> ThreadLocal::ThreadLocalObjectSharedPtr {
     return std::make_shared<TlsCache>();
   });
+
+  // Initialize stats for each thread's dispatcher. This can't be done in the InitializeCb passed
```

Comment:
Thoughts on whether this is the right place to do this? Should we instead be initializing worker stats in the server or listener manager when we create the workers? It feels that might be the right place vs. backing this into the store itself? WDYT?

Reply:
Good question. The way my initial draft implementation worked was, the `DispatcherStats` object for each `DispatcherImpl` was initialized in its constructor, rather than allocated on the heap in `initializeStats()`. I still needed to have a separate `initializeStats()` method, though, to indicate when it's safe to start actually writing to them, because the workers are allocated before `TLS::initializeThreading()` is called (meaning there's no slot to write to yet).

The main thing I didn't like about that approach was just that it required plumbing the parent `Stats::Scope` into `DispatcherImpl`'s ctor, which ended up touching a lot of stuff (e.g. `ValidationDispatcher`, `Api::allocateDispatcher()`, etc.).

Having said that, it still might be the right way to go, if we think it's useful for all dispatchers (not just the main and worker threads) to have these stats.

Reply:
Sorry I think what I'm proposing is what you are doing, but not initializing dispatcher stats inside the thread local store which seems not quite right to me. I would probably actually do this at the server level after the workers are created and `initializeThreading` is called. Then you could iterate through all workers and initialize dispatcher stats on the worker. I think this would make it a lot more clear that we are only doing this on workers, and then in the future we could do it on other dispatchers if we want to.

Reply:
Yeah, I think I understand what you're suggesting. The workers are all managed by the listener manager, so I'll have to do a little plumbing there, but that shouldn't be bad.

Reply:
Ok, plumbing is added in this revision. Stats are now initialized for each worker in `ListenerManagerImpl::startWorkers`, which seems more appropriate.